File: DSA_python/DSA06046 CHÊNH LỆCH NHỎ NHẤT.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    A.sort()

    mini = 99999999999
    for i in range(n-1):
        if mini > A[i+1] - A[i]:
            mini = A[i+1] - A[i]
    
    print(mini)


test = int(input())
for i in range(test):
    submit()

File: DSA/DSA06046 CHÊNH LỆCH NHỎ NHẤT.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;
    
    sort(A.begin(), A.end());

    int mini = INT32_MAX;

    for (int i = 0; i < n-1; i++)
    {
        if (mini > A[i+1] - A[i])
        {
            mini = A[i+1] - A[i];
        }
    }
    
    cout << mini;

    cout << endl;
}


int main()
{
    ios::sync_with_stdio(false);                                    // Speed down input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}